app/services/authn_context_resolver.rb

```diff
@@ -54,18 +66,51 @@ def no_identity_proofing_vot
     end
   end
 
+  # Returns a copy of acr_values where the appropriate IAL authentication context class reference
+  # name replaces <tt>Saml::Idp::Constants::IAL2_BIO_PREFERRED_AUTHN_CONTEXT_CLASSREF</tt>.
+  # @note This implementation will be replaced with a more robust long-term
+  # solution.
+  # @param [String] acr_values
+  def initialize_acr_values(acr_values)
+    if sp_in_biometric_pilot? && acr_values.present?
+      biometric_proofing_requested = acr_values.include?(Saml::Idp::Constants::IAL2_BIO_PREFERRED_AUTHN_CONTEXT_CLASSREF)
+      if biometric_proofing_requested && user&.identity_verified_with_biometric_comparison?
+        acr_values.gsub(
+          Saml::Idp::Constants::IAL2_BIO_PREFERRED_AUTHN_CONTEXT_CLASSREF,
+          Saml::Idp::Constants::IAL2_BIO_REQUIRED_AUTHN_CONTEXT_CLASSREF
+        )
+      elsif biometric_proofing_requested && user&.identity_verified?
+        acr_values.gsub(
+          Saml::Idp::Constants::IAL2_BIO_PREFERRED_AUTHN_CONTEXT_CLASSREF,
+          Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
+        )
+      end
```

Comment:
i think this is not quite right -- there's no condition here for acr_values.present? and not biometric.

```
return nil unless acr_values.present?

      biometric_proofing_requested = ...
      if biometric_proofing_requested && user&.identity_verified_with_biometric_comparison?
        ...
      elsif biometric_proofing_requested && user&.identity_verified?
        ...
      else
        acr_values
      end
```